paimon-core/src/main/java/org/apache/paimon/operation/AbstractFileStoreWrite.java

```diff
@@ -266,6 +246,37 @@ public List<CommitMessage> prepareCommit(boolean waitCompaction, long commitIden
         return result;
     }
 
+    protected abstract Function<WriterContainer<T>, Boolean> createWriterCleanChecker();
```

Comment:
Should we do this abstraction? Or just introducing a `boolean doWriterCleanChecker`?

Reply:
This abstract function returns a whole function (instead of just a boolean value), because we do not want to introduce `commitUser` into this base class.

For writers with no conflicts, `commitUser` might be some random value.